modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java

```diff
@@ -163,6 +165,34 @@ private static boolean shouldRunInLegacyMode(String version) {
         return true;
     }
 
+    /**
+     * Provides a docker argument string including all default labels set on testcontainer containers
+     * @return Argument string in the format `-l key1=value1 -l key2=value2`
+     */
+    private static String internalMarkerLabels() {
+        return Stream
+            .concat(
+                DockerClientFactory.DEFAULT_LABELS.entrySet().stream(),
+                ResourceReaper.instance().getLabels().entrySet().stream()
+            )
+            .map(entry -> String.format("-l %s=%s", entry.getKey(), entry.getValue()))
+            .collect(Collectors.joining(" "));
+    }
+
+    /**
+     * Configure the LocalStack container to include the default testcontainer labels on all spawned lambda containers
+     * Necessary to properly clean up lambda containers even if the LocalStack container is killed before it gets the
+     * chance.
+     */
+    private void configureLambdaContainerLabels() {
+        String lambdaDockerFlags = internalMarkerLabels();
+        String existingLambdaDockerFlags = getEnvMap().get("LAMBDA_DOCKER_FLAGS");
+        if (existingLambdaDockerFlags != null) {
+            lambdaDockerFlags = existingLambdaDockerFlags + " " + lambdaDockerFlags;
+        }
+        withEnv("LAMBDA_DOCKER_FLAGS", lambdaDockerFlags);
```

Comment:
Thinking out loud: LocalStack could detect running in a container and copy its own flags (via inspect since it has access to Docker socket) to Lambda containers it creates

Reply:
That would be an option as well, I agree. However, this was not a feature ever requested by our customers, often it is sufficient to manually specify it via the various docker flags.

My personal issue with it is: Would the user expect labels to automatically propagate in all cases, and I am not too sure about that. Of course we could make this configurable as well. I will put some more thought onto that and discuss it internally.

Another thing that comes to mind is the propagation of labels of the image itself, and of e.g. docker compose, which have to be considered and maybe filtered.